creusot/tests/should_succeed/iterators/02_iter_mut/why3session.xml

```diff
@@ -42,17 +42,17 @@
   <proof prover="0"><result status="valid" time="0.007177" steps="13"/></proof>
   </goal>
   <goal name="vc_next.1" expl="type invariant" proved="true">
-  <proof prover="3"><result status="valid" time="0.123258" steps="15265"/></proof>
+  <proof prover="3" timelimit="1"><result status="valid" time="0.123258" steps="15601"/></proof>
   </goal>
   <goal name="vc_next.2" expl="postcondition" proved="true">
   <transf name="split_vc" proved="true" >
    <goal name="vc_next.2.0" expl="postcondition" proved="true">
-   <proof prover="3"><result status="valid" time="0.066377" steps="9131"/></proof>
+   <proof prover="3" timelimit="1"><result status="valid" time="0.066377" steps="9048"/></proof>
    </goal>
   </transf>
   </goal>
   <goal name="vc_next.3" expl="postcondition" proved="true">
-  <proof prover="0"><result status="valid" time="0.161973" steps="3021"/></proof>
```

Comment:
I wish we had a way to rigorously evaluate performance impact because I get the impression we significantly increased the time on this proof, no? or is that a machine / prover version artifact?

Reply:
This change in performance is caused (I beleive) by one change:

```
-    #[ensures(forall<i : _> 0 <= i && i < result.len() ==> *result[i] == self[i])]
-    #[ensures(forall<i : _> 0 <= i && i < result.len() ==> ^result[i] == (^self)[i])]
+    #[ensures(forall<i : _> 0 <= i && i < result.len() ==> result[i] == &mut self[i])]
+    // TODO: replace with a map function applied on a sequence
     fn to_mut_seq(&mut self) -> Seq<&mut T> {
```

This is only weakly related to final borrows, so I may do this in a separate PR if you prefer.

It apparently causes issues with triggers so that alt-ergo is no longer able to prove the goal, and we need to use CVC5 instead. I still think this change is worth it because:
   - The new spec is shorter and conveys more information
   - While the proof of `all_zero` in `02_iter_mut` indeed needs fixing, the one in `03_std_iterator`, which uses the slightly different version of `IterMut` present in creusot-contracts, hasn't changed.

Also note that the timings may be biased since I ran the tests on my heavily loaded desktop.

Reply:
What about with z3?

Reply:
Sorry, I was thinking about `vc_all_zero.7` bellow. Z3 finishes  on `vc_all_zero.7` in about the same time (0.5s).

As for `vc_next.3`, I'm not sure what to think. It's quite common to have one or two goals which are significantly slower like this.

Reply:
Then it's a matter of setting the right triggers for the spec of `to_mut_seq`. We don't support triggers for lemma functions currently.